src/Compilers/CSharp/Test/Semantic/SourceGeneration/AdditionalSourcesCollectionTests.cs

```diff
@@ -106,7 +106,15 @@ public void AddedSources_Are_Deterministic()
             asc = new AdditionalSourcesCollection(".cs");
             for (int i = 0; i < 1000; i++)
             {
-                names[i] = r.NextDouble().ToString() + ".cs";
+                // ensure the name is unique (we've seen CI failures where we generate duplicate names)
+                var name = "";
```

Comment:
Could we just use a GUID?

Reply:
Yes, that would work too. Although, you known, I never thought i'd see us get two random values duplicate in 1000 iterations either, but here we are.

Reply:
Consider using `TestBase.GetUniqueName()`.

Reply:
Hmm, TIL, thanks @cston

Reply:
> Yes, that would work too. Although, you known, I never thought i'd see us get two random values duplicate in 1000 iterations either, but here we are.

If we see colliding GUIDs, we have bigger problems :).